pkg/await/version.go

```diff
@@ -66,29 +67,37 @@ func parseGitVersion(gitVersion string) (serverVersion, error) {
 	if err != nil {
 		return serverVersion{}, err
 	}
+	ret.Patch, err = strconv.Atoi(parsedVersion[3])
+	if err != nil {
+		return serverVersion{}, err
+	}
 	return ret, nil
 }
 
 // parseVersion parses version.Info into a serverVersion struct
 func parseVersion(v *version.Info) (ret serverVersion, err error) {
-	ret.Major, err = strconv.Atoi(v.Major)
+	ret, err = parseGitVersion(v.GitVersion)
```

Comment:
Just for my own understanding, we're trying to parse the git version first because `version.Info` does not have a patch number, right? It seems weird to fall back to `major.minor` in the `version.Info`, because it has no patch information? It seems like (slightly!) better option to lying about having patch info would be to:

1. Not change `serverVersion` struct to have patch information.
1. Change `parseGitVersion` to return `(int, int, int, error)`
1.  `hasHeadlessServicePortBug` call `parseGitVersion`, and fall back to `parseVersion` only if that fails (rare?)
1. Have `ServerVersion` default to major/minor, and fall back to those recovered from `parseGitVersion`.
1. Create new, simple tests for `parseGitVersion` (these are currently baked into `serverVersion` tests)

Less preferred for me is:

1. Leave a comment here explaining this behavior, as well as the fact that we expect to only rarely fall back to the `major.minor` form?
1. Right next to the comment, perhaps set `ret.Patch` specifically below so that users know that we're NOT doing this when we default to this behavior?

Reply:
I think if it doesn't exist, it's on purpose, and we should choose not to depend on that information unless we _absolutely_ have to. So I'd actually prefer _not_ to default to `0` unless there is no other choice. It seems to me that if we promise we're going to return patch info in this API I think we are just waiting to shoot ourselves, for not much benefit?